pkg/library/reconcilers/base_reconciler.go

```diff
@@ -61,9 +70,14 @@ func NewBaseReconciler(
 		apiClientReader: apiClientReader,
 		logger:          logger,
 		recorder:        recorder,
+		eventHandlers:   []EventHandler{},
```

Comment:
Are we concerned about multiple registrations happening leading to possible multiple increments? Should we ensure uniqueness ?

Reply:
I am not convinced this is a good approach.
1) If you're registering the same event handler multiple times (e.g. registration during a reconcile loop), you have a need to refactor your code, and this misbehaving would highlight that - where preventing multiple registrations would mask the problem
2) It's not impossible to want multiple of the same type of event handler, but that are configured differently